### PR TITLE
[SDP-877] Add `signature-service-type` config option that can be used to configure which signature service implementation the host wants to use

### PR DESCRIPTION
**Pull request by marcelosalloum**
### What

Add `signature-service-type` config option that can be used to configure which signature service implementation the host wants to use.

Also, update the error messages from the custom_set_value.go methods, so they convey the information about which config option was rejected.

### Why

As part of https://stellarorg.atlassian.net/browse/SDP-877

#### PR Structure

* [x] This PR has a reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR title and description are clear enough for anyone to review it.
* [ ] This PR does not mix refactoring changes with feature changes (split into two PRs otherwise).

#### Thoroughness

* [x] This PR adds tests for the new functionality or fixes.
* [x] This PR contains the link to the Jira ticket it addresses.

#### Configs and Secrets

* [x] No new CONFIG variables are required -OR- the new required ones were added to the helmchart's [`values.yaml`] file.
* [x] No new CONFIG variables are required -OR- the new required ones were added to the deployments ([`pr-preview`], [`dev`], [`demo`], `prd`).
* [x] No new SECRETS variables are required -OR- the new required ones were mentioned in the helmchart's [`values.yaml`] file.
* [x] No new SECRETS variables are required -OR- the new required ones were added to the deployments ([`pr-preview secrets`], [`dev secrets`], [`demo secrets`], `prd secrets`).

#### Release

* [x] This is not a breaking change.
* [x] **This is ready for production.**. If your PR is not ready for production, please consider opening additional complementary PRs using this one as the base. Only merge this into `develop` or `main` after it's ready for production!

#### Deployment

* [ ] Does the deployment work after merging?

[`values.yaml`]: ../helmchart/sdp/values.yaml
[`pr-preview`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/stellar-disbursement-platform/backend-helm-values
[`dev`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/backend-helm-values
[`demo`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-backend-helm-values
[`pr-preview secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/common-previews/externalsecrets-common-previews.yaml#L241-L346
[`dev secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/stellar-disbursement-platform-externalsecrets.yaml
[`demo secrets`]: https://github.com/stellar/kube/blob/d3e4f5dd8aa4c13b45a31a5a937f3e98841171a7/kube001-dev/namespaces/stellar-disbursement-platform/demo/demo-sdp-externalsecrets.yaml

### CODE REVIEW

File: .github/workflows/e2e_integration_test.yml [hunk 1]
```diff
@@ -25,6 +25,7 @@ jobs:
     env:
       DISTRIBUTION_PUBLIC_KEY: ${{ vars.DISTRIBUTION_PUBLIC_KEY }}
       DISTRIBUTION_SEED: ${{ vars.DISTRIBUTION_SEED }}
+      CHANNEL_ACCOUNT_ENCRYPTION_PASSPHRASE: ${{ vars.DISTRIBUTION_SEED }}
```
Comment (CaioTeixeira95) [inline .github/workflows/e2e_integration_test.yml:28]: Wasn't this being added on the last PR?

File: internal/integrationtests/docker-compose-e2e-tests.yml [hunk 2]
```diff
@@ -101,6 +102,7 @@ services:
       TSS_METRICS_PORT: "9002"
       TSS_METRICS_TYPE: "TSS_PROMETHEUS"
       DISTRIBUTION_SEED: ${DISTRIBUTION_SEED}
+      EVENT_BROKER_TYPE: "NONE"
```
Comment (CaioTeixeira95) [inline internal/integrationtests/docker-compose-e2e-tests.yml:105]: Thank you adding this.

**Review by CaioTeixeira95 — APPROVED** [review]
LGTM

File: .github/workflows/e2e_integration_test.yml (hunk 1 above)
Comment (marcelosalloum) [reply .github/workflows/e2e_integration_test.yml:28]: This addition was missing in the e2e docker compose file.

**Pull request merged**